pkg/resource/db_proxy/hooks.go

```diff
@@ -25,6 +28,86 @@ import (
 	"github.com/aws-controllers-k8s/rds-controller/pkg/util"
 )
 
+var (
+	// TerminalStatuses are the status strings that are terminal states for a
+	// DB proxy.
+	TerminalStatuses = []string{
+		svcsdk.DBProxyStatusIncompatibleNetwork,
+		svcsdk.DBProxyStatusDeleting,
+		svcsdk.DBProxyStatusSuspending,
```

Comment:
Are these terminal statuses though? These aren't statuses that would require the user to change something in the `Spec` so probably should not be terminal conditions...

Reply:
`Deleting` is a terminal status for `DBInstance` - https://github.com/aws-controllers-k8s/rds-controller/blob/main/pkg/resource/db_instance/hooks.go#L89

As for `Suspending`, I figured it was inline with `Deleting` as a status that users could not get out from by modifying spec.

Reply:
Hmmm, yeah. I guess it's OK since this will only cause a Terminal condition to appear on the resource IFF the user has deleted or suspended a DBProxy AND THEN made a change to a Spec field, since this is only called on sdkUpdate